lib/adUser.class.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+include sfConfig::get('sf_root_dir') .'/vendor/password_compat/password.php';
+
+class adUser extends myUser implements Zend_Acl_Role_Interface
+{
+  protected $ldapConnection;
+  protected $ldapBound;
+
+  public function initialize(sfEventDispatcher $dispatcher, sfStorage $storage, $options = array())
+  {
+    // initialize parent
+    parent::initialize($dispatcher, $storage, $options);
+
+    if (!extension_loaded('ldap'))
+    {
+      throw new sfConfigurationException('ldapUser class needs the "ldap" extension to be loaded.');
+    }
+  }
+
+  public function authenticate($username, $password)
+  {
+    // Allow LDAP authentication to be overridden during development
+    $configuration = sfContext::getInstance()->getConfiguration();
+    if ($configuration->isDebug() || 'dev' == $configuration->getEnvironment())
+    {
+      return parent::authenticate($username, $password);
+    }
+
+    // Anonymous is not a real user
+    if ($username == 'anonymous')
+    {
+      return false;
+    }
+
+    $authenticated = $this->ldapAuthenticate($username, $password);
+
+    // Fallback to non-LDAP authentication if need be and load/create user data
+    if (!$authenticated)
+    {
+      $authenticated = parent::authenticate($username, $password);
+
+      // Load user
+      $criteria = new Criteria;
+      $criteria->add(QubitUser::EMAIL, $username);
+      $user = QubitUser::getOne($criteria);
+    }
+    else
+    {
+      // Load user using username or, if one doesn't exist, create it
+      $criteria = new Criteria;
+      //$criteria->add(QubitUser::USERNAME, $username);
```

Comment:
Probably best to remove these two commented-out lines.

Reply:
Removed